netdata-installer.sh

```diff
@@ -1108,21 +1148,92 @@ NETDATA_GROUP="$(id -g -n "${NETDATA_USER}" 2> /dev/null)"
 [ -z "${NETDATA_GROUP}" ] && NETDATA_GROUP="${NETDATA_USER}"
 echo >&2 "Netdata user and group set to: ${NETDATA_USER}/${NETDATA_GROUP}"
 
+NETDATA_CMAKE_OPTIONS="-S ./ -B ${NETDATA_BUILD_DIR} ${ninja:+-G Ninja} -DCMAKE_INSTALL_PREFIX=${NETDATA_PREFIX} ${NETDATA_USER:+-DNETDATA_USER=${NETDATA_USER}} "
+
+# Feature autodetection code starts here
+
+if [ "${USE_SYSTEM_PROTOBUF}" -eq 1 ]; then
+  NETDATA_CMAKE_OPTIONS="${NETDATA_CMAKE_OPTIONS} -DENABLE_BUNDLED_PROTOBUF=Off"
+else
+  NETDATA_CMAKE_OPTIONS="${NETDATA_CMAKE_OPTIONS} -DENABLE_BUNDLED_PROTOBUF=On"
+fi
+
+if [ -z "${ENABLE_SYSTEMD_PLUGIN}" ]; then
+    if check_for_module libsystemd; then
+        if check_for_module libelogind; then
+            ENABLE_SYSTEMD_JOURNAL=0
+        else
+            ENABLE_SYSTEMD_JOURNAL=1
+        fi
+    else
+        ENABLE_SYSTEMD_JOURNAL=0
+    fi
+fi
+
+enable_feature PLUGIN_SYSTEMD_JOURNAL "${ENABLE_SYSTEMD_JOURNAL}"
+
+[ -z "${NETDATA_ENABLE_ML}" ] && NETDATA_ENABLE_ML=1
+enable_feature ML "${NETDATA_ENABLE_ML}"
+
+if command -v cups-config >/dev/null 2>&1 || check_for_module libcups || check_for_module cups; then
```

Comment:
@Ferroin This check will return true on centos 7.

The cups plugin relies on API version >= 1.7. You can use `cups-config --api-version` to get the version.